onward/app/feed/MostPopularAgent.scala

```diff
@@ -23,25 +28,89 @@ object MostPopularRefresh {
+class MostPopularAgent(contentApiClient: ContentApiClient, ophanApi: OphanApi, wsClient: WSClient) extends Logging {
 
   private val agent = Box[Map[String, Seq[RelatedContentItem]]](Map.empty)
 
+  // Helper case class to read from the most/comments discussion API call.
+  private case class MostDiscussedItem(key: String, url: String, numberOfComments: Int)
+  private object MostDiscussedItem {
+    implicit val format = Json.format[MostDiscussedItem]
+  }
+
+  // Container for most_shared and most_commented
+  val mostSingleCards = Box[Map[String,Content]](Map.empty)
+
   def mostPopular(edition: Edition): Seq[RelatedContentItem] = agent().getOrElse(edition.id, Nil)
 
-  def refresh()(implicit ec: ExecutionContext): Future[Map[String, Seq[RelatedContentItem]]] = {
-    log.info("Refreshing most popular.")
+  // Note that here we are in procedural land here (not functional)
+  def refresh()(implicit ec: ExecutionContext): Unit = {
     MostPopularRefresh.all(Edition.all)(refresh)
+    refreshGlobal()
   }
 
-  private def refresh(edition: Edition)(implicit ec: ExecutionContext): Future[Map[String, Seq[RelatedContentItem]]] =
-    contentApiClient.getResponse(contentApiClient.item("/", edition)
-      .showMostViewed(true)
-    ).flatMap { response =>
-      val mostViewed = response.mostViewed.getOrElse(Nil).take(10).map(RelatedContentItem(_))
-      agent.alter(_ + (edition.id -> mostViewed))
+  private def refreshGlobal()(implicit ec: ExecutionContext): Future[Map[String,Content]] = {
+
+    log.info("Pulling most social media shared from Ophan")
+
+    val sinceHours = 3
+    val sinceTimestamp = System.currentTimeMillis - sinceHours * 60 * 60 * 1000
+
+    val futureMostFaceBook = ophanApi.getMostReadFacebook(sinceHours)
+    val futureMostCommented = mostCommented(wsClient, sinceTimestamp)
+
+    for {
+        mostFacebook <- futureMostFaceBook
+        oneFacebookMostRead = mostFacebook.headOption.get
+        oneFacebookContent <- contentFromUrl(oneFacebookMostRead.url, contentApiClient)
+        _ <- mostSingleCards.alter(_ + ("most_shared" -> oneFacebookContent))
+
+        oneMostCommentedItem <- futureMostCommented
+        oneMostCommentedContent <- contentFromUrl(oneMostCommentedItem.url, contentApiClient)
+        newMap <- mostSingleCards.alter( _ + ("most_commented" -> oneMostCommentedContent))
+      } yield newMap
+  }
+
+  private def mostCommented(wsClient: WSClient, since: Long)(implicit ec: ExecutionContext): Future[MostDiscussedItem] = {
+    val dapiURL = Configuration.discussion.apiRoot
+    val params = List("api-key" -> "dotcom", "pageSize" -> "10", "sinceTimestamp" -> since.toString )
+
+    val fResponse = wsClient.url(dapiURL + "/most/comments")
+      .addQueryStringParameters(params: _*)
+      .get()
+
+    fResponse.map{ r =>
+      val json = r.json
+      ( json \ "discussions" ).as[List[MostDiscussedItem]]
+        // Avoiding live blogs
+        .filter{ item => ! item.url.contains("/live/") }
```

Comment:
Could be `filterNot`.  And maybe extract function so don't need comment.
Eg. `filterNot { isLiveBlog }`

Reply:
shoulds good 👍